src/sas/__init__.py

```diff
@@ -12,10 +16,59 @@ def get_app_dir():
     """
     global _APP_DIR
     if not _APP_DIR:
-        from ._config import find_app_dir
         _APP_DIR = find_app_dir()
     return _APP_DIR
 
+# TODO: Replace with more idomatic version
+def dirn(path, n):
+    """
+    Return the directory n up from the current path
+    """
+    path = os.path.realpath(path)
+    for _ in range(n):
+        path = os.path.dirname(path)
+    return path
+
+
+def find_app_dir():
+    """
+    Locate the parent directory of the sasview resources.  For the normal
+    application this will be the directory containing sasview.py.  For the
+    frozen application this will be the path where the resources are installed.
+    """
+    # We are starting out with the following info:
+    #     __file__ = .../sas/__init__.pyc
+    # Check if the path .../sas/sasview exists, and use it as the
+    # app directory.  This will only be the case if the app is not frozen.
+    path = os.path.join(os.path.dirname(__file__), 'sasview')
+    if os.path.exists(path):
+        return path
+
+    # If we are running frozen, then root is a parent directory
+    if sys.platform == 'darwin':
+        # Here is the path to the file on the mac:
+        #     .../Sasview.app/Contents/Resources/lib/python2.7/site-packages.zip/sas/__init__.pyc
+        # We want the path to the Resources directory.
+        path = dirn(__file__, 5)
+    elif os.name == 'nt':
+        # Here is the path to the file on windows:
+        #     ../Sasview/library.zip/sas/__init__.pyc
+        # We want the path to the Sasview directory.
+        path = dirn(__file__, 3)
```

Comment:
The depths here are wrong as of v5.0.4 as outlined in #1911. I was looking for a cross-platform solution to this in #2171 using `glob` but wasn't able to 100% solve the issue.

Reply:
generally, we should be using `importlib.resources` where possible. This is on the list of things to do, but quite a long way down